network-libp2p/src/discovery/peer_contacts.rs

```diff
@@ -509,11 +666,15 @@ impl PeerContactBook {
 
     /// Gets a set of peer contacts given a services filter.
     /// Every peer contact that matches such services will be returned.
-    pub fn query(&self, services: Services) -> impl Iterator<Item = Arc<PeerContactInfo>> + '_ {
+    pub fn query(
+        &self,
+        services: Services,
+        include_local_only: bool,
+    ) -> impl Iterator<Item = Arc<PeerContactInfo>> + '_ {
         // TODO: This is a naive implementation
         // TODO: Sort by score?
         self.peer_contacts.iter().filter_map(move |(_, contact)| {
-            if contact.matches(services) {
+            if !contact.contact.local_only || include_local_only && contact.matches(services) {
```

Comment:
Is this what we want? If a peer is not `local_only`, we do not filter by services?
The current code is equivalent to `!contact.contact.local_only || (include_local_only && contact.matches(services))`